Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/StructurePanel.java

```diff
@@ -719,34 +722,75 @@ private boolean treeNodeMatchesStructure(IncludedStructuralElement structure, Tr
      */
     public void onDragDrop(TreeDragDropEvent event) {
 
-        TreeNode dragTreeNode = event.getDragNode();
-        TreeNode dropTreeNode = event.getDropNode();
-
-        Object dragNodeObject = dragTreeNode.getData();
-        Object dropNodeObject = dropTreeNode.getData();
-
-        if (!(dragNodeObject instanceof StructureTreeNode) || !(dropNodeObject instanceof StructureTreeNode)) {
-            Helper.setErrorMessage(Helper.getTranslation("dataEditor.unableToMoveError"));
-            return;
-        }
-        StructureTreeNode dropNode = (StructureTreeNode) dropNodeObject;
-        StructureTreeNode dragNode = (StructureTreeNode) dragNodeObject;
-
+        Object dragNodeObject = event.getDragNode().getData();
+        Object dropNodeObject = event.getDropNode().getData();
 
         try {
+            StructureTreeNode dropNode = (StructureTreeNode) dropNodeObject;
+            StructureTreeNode dragNode = (StructureTreeNode) dragNodeObject;
             if (dragNode.getDataObject() instanceof IncludedStructuralElement
                     && dropNode.getDataObject() instanceof IncludedStructuralElement) {
                 checkLogicalDragDrop(dragNode, dropNode);
             } else if (dragNode.getDataObject() instanceof MediaUnit
                     && dropNode.getDataObject() instanceof MediaUnit) {
                 checkPhysicalDragDrop(dragNode, dropNode);
+            } else if (dragNode.getDataObject() instanceof  View && dropNode.getDataObject()
+                    instanceof IncludedStructuralElement) {
+                TreeNode dragParent = event.getDragNode().getParent();
+                if (dragParent.getData() instanceof StructureTreeNode) {
+                    StructureTreeNode dragParentTreeNode = (StructureTreeNode) dragParent.getData();
+                    if (dragParentTreeNode.getDataObject() instanceof IncludedStructuralElement) {
+                        View view = (View) dragNode.getDataObject();
+                        IncludedStructuralElement previousParent;
+                        if (dataEditor.getWorkpiece().getRootElement().getViews().contains(view)) {
+                            previousParent = dataEditor.getWorkpiece().getRootElement();
+                        } else {
+                            previousParent = getPageStructure(view, dataEditor.getWorkpiece().getRootElement());
+                        }
+                        if (Objects.nonNull(previousParent)) {
+                            IncludedStructuralElement element = (IncludedStructuralElement) dropNode.getDataObject();
+                            List<View> newViews = new LinkedList<>(element.getViews());
+                            // TODO: decide whether the order of pages inside an element should be changeable!
```

Comment:
I thought we decided on that!?

Reply:
Yes, you are right. I think I opened this PR before we made this decision, though. I will close the PR and re-open it once I have implemented the required changes.